d2bs/kolbot/libs/OOG.js

```diff
@@ -693,7 +675,7 @@ const ControlAction = {
 		MainLoop:
 		while (true) {
 			switch (getLocation()) {
-			case 8:
+			case sdk.game.locations.MainMenu:
 				control = getControl(6, 264, 391, 272, 25);
```

Comment:
Is this the reason for naming the helper module Control_1?
The way I see it you should be able to also refactor these calls by:
```js
let con = Control_1.Gateway.control();
```

or maybe just directly access the properties `let realm_str = Control_1.Gateway.text;` in cases where you don't reuse the reference? It should return null in case it isn't there or control isn't found.

Reply:
It refetches the control every time so it makes it more safe, as it happens to disappear from time to.

However, yeah, the Control_1 better be named Control. This might come from one of my private repos where typescript compiles it to Control_1

Reply:
Yeah in that portion I picked to comment it only reads the text property of the control and then does some language specific manipulation on it, it shouldn't matter if the control itself vanishes in between if you already fetched the text property. Also, can it really as long as there's no sort of yield/delay?!

So you basically say a 'global' reference to the control shouldn't be held in any case because you cannot rely on it. I think the only reason for `let control` at the function entry is because the switch case's aren't scoped.

I really like the idea of getting rid of these magic positions spread all over the scripts. Just realised some where still there.

How about another method `exists()` for your wrapper?

Reply:
hmm yeah that s a good point, rename it to Controls eh

Reply:
But since it's a singleton you could account for that no?

Reply:
```
let con = Control_1.CharSelectCharInfo0.control();
```
doesn't work
EDIT: needed to be
```
let con = Control_1.CharSelectCharInfo0.control
```